drivers/charger/sbs_charger/sbs_charger.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_SBS_CHARGER_H_
+#define ZEPHYR_DRIVERS_SENSOR_SBS_CHARGER_H_
+
+#include <stdint.h>
+#include <zephyr/drivers/i2c.h>
+
+#define SBS_CHARGER_REG_SPEC_INFO     0x11
+#define SBS_CHARGER_REG_STATUS	      0x13
+#define SBS_CHARGER_REG_ALARM_WARNING 0x16
+
+#define SBS_CHARGER_STATUS_CHARGE_INHIBITED BIT(1)
+#define SBS_CHARGER_STATUS_RES_COLD	    BIT(9)
+#define SBS_CHARGER_STATUS_RES_HOT	    BIT(10)
+#define SBS_CHARGER_STATUS_BATTERY_PRESENT  BIT(14)
+#define SBS_CHARGER_STATUS_AC_PRESENT	    BIT(15)
+
+#define SBS_CHARGER_POLL_TIME 500
+
+struct sbs_charger_config {
+	struct i2c_dt_spec i2c;
+};
```

Comment:
why do we need an internal header? is this shared with other modules?

Reply:
This header is specific to an SBS compliant charging IC. The spec was last ratified in 1998, so charging IC vendors have moved past this basic specification.

The intention is to support charging ICs of various vendors. The SBS charger does have some relevance to this day and serves a pretty basic example to develop this API against.

I'm not sure if this totally answers your questions, please let me know if I can clarify further.